ipld/resolver/src/behaviour/discovery.rs

```diff
@@ -115,6 +115,10 @@ impl Behaviour {
             let protocol_name = format!("/ipc/{}/kad/1.0.0", nc.network_name);
             kad_config.set_protocol_names(vec![Cow::Owned(protocol_name.as_bytes().to_vec())]);
 
+            // Disable inserting records into the memory store, so peers cannot send `PutRecord`
+            // messages to store content in the memory of our node.
+            kad_config.set_record_filtering(KademliaStoreInserts::FilterBoth);
```

Comment:
Just a sanity-check (I haven't looked in-depth to the rust impl): disabling `StoreInserts` still allow peers to advertise their contact information on a `PeerRecord` right? If not this may break our discovery service.

Reply:
It seems to work. `find_nearest_peers`, `bootstrap`, `AddProvider` and `PutRecord` all enqueue background queries to find the `ClosestPeers` to the one in the record, so they have the same effect. It looks to me like if we don't want to advertise nodes providing specific content, then we can disable the store completely.